src/components/common/ColonyActions/helpers/mapItemToMessageFormat.tsx

```diff
@@ -235,5 +237,50 @@ export const mapColonyActionToExpectedFormat = (
         ActionTitleMessageKeys.SafeTransactionTitle,
         notMaybe(actionData.metadata?.customTitle),
       ),
+    [ActionTitleMessageKeys.RecipientsNumber]: expenditureData?.slots?.reduce(
+      (
+        result: {
+          recipients: string[];
+        },
+        slot,
+      ) => {
+        const { recipientAddress } = slot;
+
+        if (
+          typeof recipientAddress === 'string' &&
+          !result.recipients.includes(recipientAddress)
+        ) {
+          result.recipients.push(recipientAddress);
+        }
+
+        return result;
+      },
+      {
+        recipients: [],
+      },
+    ).recipients.length,
+    [ActionTitleMessageKeys.TokensNumber]: expenditureData?.slots?.reduce(
+      (
+        result: {
+          tokens: string[];
+        },
+        slot,
+      ) => {
+        const { payouts } = slot;
+        const tokenAddress = payouts?.[0]?.tokenAddress;
```

Comment:
In the UI it's not possible to have more than one payout per slot, but we have to account for that possibility on the contracts side. 

Could also use a set with a `flatMap` for that:

```ts
[ActionTitleMessageKeys.TokensNumber]: new Set(
  expenditureData?.slots?.flatMap(
    (slot) => slot.payouts?.map((payout) => payout.tokenAddress) ?? [],
  ),
).size,
```